src/zookeeper/connection.go

```diff
@@ -226,18 +250,21 @@ func GetBrokerConnectionInfo(brokerID int, zkConn Connection) (scheme, brokerHos
 	}
 
 	// We only want the URL if it's SSL or PLAINTEXT
-	scheme, brokerURL, err := getURLStringAndSchemeFromEndpoints(brokerDecoded.Endpoints, brokerDecoded.ProtocolMap)
+	schemes, brokerURLs, err := getURLStringAndSchemeFromEndpoints(brokerDecoded.Endpoints, brokerDecoded.ProtocolMap)
 
 	if err != nil {
 		return
 	}
 
-	host, portString := brokerURL.Hostname(), brokerURL.Port()
+	connections := make([]BrokerConnection, 0)
```

Comment:
Rather than capacity 0, capacity should be set to `len(brokerURLs)` to reduce allocations. Super minor, but 🤷‍♂